### PR TITLE
Book: Small grammatical and stylistic edits to book

### PR DESCRIPTION
**Pull request by jonmarkprice**
I've been reading [the book](https://doc.rust-lang.org/book/) and noticed a few small grammatical and stylistic issues which I've rolled into this pull request. 

I'm not sure if I should do so many small, unrelated edits in a single pull request but it seems like a lot of overhead for each small edit. Maybe one commit per edit but one pull request per file/section? Feedback is very much appreciated as this is my first pull request ever!

r? @steveklabnik rollup

### CODE REVIEW

**Comment by steveklabnik** [comment]
> Feedback is very much appreciated as this is my first pull request ever!

Thank you! It's not super important; one commit per section or one commit for the whole thing is fine.

File: src/doc/book/ownership.md [hunk 1]
```diff
@@ -212,7 +212,7 @@ But, unlike a move, we can still use `v` afterward. This is because an `i32`
 has no pointers to data somewhere else, copying it is a full copy.
 
 All primitive types implement the `Copy` trait and their ownership is
-therefore not moved like one would assume, following the ´ownership rules´.
+therefore not moved like one would assume, following the 'ownership rules'.
```
Comment (steveklabnik) [inline src/doc/book/ownership.md]: we use curly quotes throughout the book

File: src/doc/book/references-and-borrowing.md [hunk 2]
```diff
@@ -208,12 +208,14 @@ With this in mind, let’s consider our example again.
 Here’s the code:
 
 ```rust,ignore
-let mut x = 5;
-let y = &mut x;
+fn main() {
```
Comment (steveklabnik) [inline src/doc/book/references-and-borrowing.md]: why are we adding main here?

**Comment by steveklabnik** [comment]
Thanks so much for this! A few nits and questions, but great overall :+1:

**Comment by jonmarkprice** [comment]
Thanks, my bad on the curly quotes. I'll fix that.

The reason I added the `main` was because the book shows an error message which refers to it:
```text
note: previous borrow ends here
fn main() {

}
^
```
but the code fragment in the book leaves it out: 
```rust
let mut x = 5;
let y = &mut x;

*y += 1;

println!("{}", x);Run
```
I thought it would be clearer if the code fragment had the main function in this case because it is referenced. But obviously this is subjective.

**Comment by steveklabnik** [comment]
> The reason I added the main was because the book shows an error message which refers to it:

Ah, I like it! Let's keep that change then :+1:

**Comment by jonmarkprice** [comment]
Ok I think I fixed the quotes. This is how it currently is in the [book](https://doc.rust-lang.org/book/ownership.html)
> ... the ´ownership rules´. To ...

compared to different quote:
> ... they ‘have ownership’ of ...

So I copied that character from the second quote and put it in the markdown, it looks weird in the Github editor though...

**Comment by steveklabnik** [comment]
@bors: r+ rollup

looks good to me!

**Comment by jonmarkprice** [comment]
> Synchronizing submodule url for 'src/rust-installer'
> fatal: Needed a single revision
> 
> command did not execute successfully: "git" "submodule" "update"
> expected success, got: exit code: 1
>
> Makefile:23: recipe for target 'all' failed
> rustjob: found 0 remaining processes
> Unable to find current revision in submodule path 'src/llvm'
> make: *** [all] Error 1

@steveklabnik any idea why the build would fail?

**Comment by alexcrichton** [comment]
@bors: retry

* https://github.com/rust-lang/rust/issues/34595

**Comment by steveklabnik** [comment]
@jonmarkprice if @alexcrichton 's comment wasn't clear, this was a failure that was 100% not your fault; a bug in our automation.

**Pull request merged**